### PR TITLE
fix: the github issue query dates weren't being updated

### LINKED ISSUE #112: The report is missing new issues
It seems that the new issues aren't being collected by the bot.

### PR DESCRIPTION
**Pull request by jacobdotcosta**
so the bot always queried issues and PRs on the same dates

Closes #112

### CODE REVIEW

File: src/main/java/io/snowdrop/github/reporting/IssueCollector.java [hunk 1]
```diff
@@ -36,30 +37,33 @@
...
     this.client = client;
     this.status = status;
     this.issueService = new IssueService(client);
     this.users = users;
     this.organizations = organizations;
+    init(Optional.of(reportingDay), Optional.of(reportingHour));
+  }
 
-    this.endTime = ZonedDateTime.now().with(DayOfWeek.of(reportingDay)).withHour(reportingHour);
+  public void init(final Optional<Integer> reportingDay, final Optional<Integer> reportingHour) {
```
Comment (iocanel) [inline src/main/java/io/snowdrop/github/reporting/IssueCollector.java]: The name init implies that it should only be called during intialization, however it's called each time we collect in order to refresh the endTime. A more fitting name is needed here.

File: src/main/java/io/snowdrop/github/reporting/IssueCollector.java [hunk 2]
```diff
@@ -68,38 +72,44 @@ public void refresh() {
   }
 
   public Map<String, Set<Issue>> collectIssues() {
+    init(Optional.empty(), Optional.empty());
```
Comment (iocanel) [inline src/main/java/io/snowdrop/github/reporting/IssueCollector.java]: It's not clear to me why we pass empty and not reportingDay, reportingTime?

File: src/main/java/io/snowdrop/github/reporting/IssueCollector.java [hunk 3]
```diff
@@ -68,38 +72,44 @@ public void refresh() {
...
+    LOGGER.info("Collecting issues: {}-{}, {}-{}", startTime, endTime, minStartTime, minEndTime);
     return streamIssues().collect(Collectors.groupingBy(Issue::getAssignee, Collectors.toSet()));
   }
+
   public Stream<Issue> streamIssues() {
+    init(Optional.empty(), Optional.empty());
```
Comment (iocanel) [inline src/main/java/io/snowdrop/github/reporting/IssueCollector.java]: It's not clear to me why we pass empty and not reportingDay, reportingTime?

File: src/main/java/io/snowdrop/github/reporting/PullRequestCollector.java [hunk 4]
```diff
@@ -61,15 +62,18 @@ public PullRequestCollector(GitHubClient client, StatusLogger status, int report
     this.users = users;
     this.organizations = organizations;
 
-    this.endTime = ZonedDateTime.now().with(DayOfWeek.of(reportingDay)).withHour(reportingHour);
-    this.startTime = endTime.minusWeeks(1);
+    init(Optional.of(reportingDay),Optional.of(reportingHour));
+  }
 
+  public void init(final Optional<Integer> reportingDay, final Optional<Integer> reportingHour) {
```
Comment (iocanel) [inline src/main/java/io/snowdrop/github/reporting/PullRequestCollector.java]: My naming objection also applies here.

File: src/main/java/io/snowdrop/github/reporting/PullRequestCollector.java [hunk 5]
```diff
@@ -81,11 +85,13 @@ public void refresh() {
   }
 
   public Map<String, Set<PullRequest>> collectPullRequests() {
+    init(Optional.empty(), Optional.empty());
```
Comment (iocanel) [inline src/main/java/io/snowdrop/github/reporting/PullRequestCollector.java]: It's not clear to me why we pass empty and not reportingDay, reportingTime?

File: src/main/java/io/snowdrop/github/reporting/PullRequestCollector.java [hunk 6]
```diff
@@ -81,11 +85,13 @@ public void refresh() {
...
     return streamPullRequests()
       .collect(Collectors.groupingBy(PullRequest::getCreator, Collectors.toSet()));
   }
 
   public Stream<PullRequest> streamPullRequests() {
+    init(Optional.empty(), Optional.empty());
```
Comment (iocanel) [inline src/main/java/io/snowdrop/github/reporting/PullRequestCollector.java]: It's not clear to me why we pass empty and not reportingDay, reportingTime?

**Review by iocanel — COMMENTED** [review]
I have some questions and some minor comments on naming.

File: src/main/java/io/snowdrop/github/reporting/IssueCollector.java (hunk 2 above)
Comment (jacobdotcosta) [reply src/main/java/io/snowdrop/github/reporting/IssueCollector.java]: So they're obtained from the system date, using the same method.

File: src/main/java/io/snowdrop/github/reporting/IssueCollector.java (hunk 1 above)
Comment (jacobdotcosta) [reply src/main/java/io/snowdrop/github/reporting/IssueCollector.java]: I'll add a better name.

File: src/main/java/io/snowdrop/github/reporting/IssueCollector.java (hunk 2 above)
Comment (iocanel) [reply src/main/java/io/snowdrop/github/reporting/IssueCollector.java]: Still dont understand this.

File: src/main/java/io/snowdrop/github/reporting/IssueCollector.java [hunk 7]
```diff
@@ -6,6 +6,7 @@
 import java.time.ZonedDateTime;
 import java.util.Date;
 import java.util.Map;
+import java.util.Optional;
```
Comment (iocanel) [inline src/main/java/io/snowdrop/github/reporting/IssueCollector.java]: This is no longer used.

**Review by iocanel — APPROVED** [review]

**Pull request merged**